extensions/rocksdb-repos/ProvenanceRepository.h

```diff
@@ -83,12 +83,11 @@ class ProvenanceRepository : public core::Repository, public std::enable_shared_
     }
     logger_->log_debug("MiNiFi Provenance Max Partition Bytes %d", max_partition_bytes_);
     if (config->get(Configure::nifi_provenance_repository_max_storage_time, value)) {
-      core::TimeUnit unit;
-      if (core::Property::StringToTime(value, max_partition_millis_, unit)) {
-        core::Property::ConvertTimeUnitToMS(max_partition_millis_, unit, max_partition_millis_);
-      }
+      auto max_partition = utils::timeutils::StringToDuration<std::chrono::milliseconds>(value);
+      if (max_partition.has_value())
```

Comment:
we could use the declaration form of the if statement here as well

Reply:
good idea, changed in https://github.com/apache/nifi-minifi-cpp/pull/1225/commits/1fcbcc05ce27c9f7386c8f5b3a4a5267593c8fa1